src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/AtomPub10CategoriesDocumentFormatter.cs

```diff
@@ -76,100 +59,75 @@ public AtomPub10CategoriesDocumentFormatter(CategoriesDocument documentToWrite)
-            TraceCategoriesDocumentReadEnd();
         }
 
         [SuppressMessage("Microsoft.Design", "CA1033:InterfaceMethodsShouldBeCallableByChildTypes", Justification = "The IXmlSerializable implementation is only for exposing under WCF DataContractSerializer. The funcionality is exposed to derived class through the ReadFrom\\WriteTo methods")]
         void IXmlSerializable.WriteXml(XmlWriter writer)
         {
             if (writer == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("writer");
+                throw new ArgumentNullException(nameof(writer));
             }
-            if (this.Document == null)
+            if (Document == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.DocumentFormatterDoesNotHaveDocument)));
+                throw new InvalidOperationException(SR.Format(SR.DocumentFormatterDoesNotHaveDocument));
             }
-            TraceCategoriesDocumentWriteBegin();
+
             WriteDocument(writer);
-            TraceCategoriesDocumentWriteEnd();
         }
 
         public override void ReadFrom(XmlReader reader)
         {
             if (reader == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("reader");
+                throw new ArgumentNullException(nameof(reader));
             }
             if (!CanRead(reader))
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new XmlException(SR.Format(SR.UnknownDocumentXml, reader.LocalName, reader.NamespaceURI)));
+                throw new XmlException(SR.Format(SR.UnknownDocumentXml, reader.LocalName, reader.NamespaceURI));
             }
-            TraceCategoriesDocumentReadBegin();
+
             ReadDocument(reader);
-            TraceCategoriesDocumentReadEnd();
         }
 
         public override void WriteTo(XmlWriter writer)
         {
             if (writer == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("writer");
+                throw new ArgumentNullException(nameof(writer));
             }
-            if (this.Document == null)
+            if (Document == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.DocumentFormatterDoesNotHaveDocument)));
+                throw new InvalidOperationException(SR.Format(SR.DocumentFormatterDoesNotHaveDocument));
             }
-            TraceCategoriesDocumentWriteBegin();
+            
             writer.WriteStartElement(App10Constants.Prefix, App10Constants.Categories, App10Constants.Namespace);
             WriteDocument(writer);
             writer.WriteEndElement();
-            TraceCategoriesDocumentWriteEnd();
-        }
-
-        internal static void TraceCategoriesDocumentReadBegin()
-        {
```

Comment:
I'm not sure about removing these. I know right now they are empty methods, but the JIT will optimize them away. We are missing some tracing code here which we might want to put back and going back through to put all the call sites back in through the code would be extra work later. I'll talk with the rest of the team to decide what we want to do with regards tracing, i.e. do we want to put it back in or leave it out.

Reply:
Spoke with others in the team, we don't have plans to reintroduce the tracing so removal is fine.